src/main/java/frc/robot/Robot.java

```diff
@@ -17,92 +22,73 @@
-
-  /**
-   * This function is run when the robot is first started up and should be used for any
-   * initialization code.
-   */
-  @Override
-  public void robotInit() {
-    
-    // Instantiate our RobotContainer.  This will perform all our button bindings, and put our
-    // autonomous chooser on the dashboard.
-    m_robotContainer = new RobotContainer();
-    new WheelAlign(m_robotContainer.getDrivetrain()).schedule();
-    new ResetGyro(m_robotContainer.getDrivetrain(), 2).schedule();
-  }
-
-  /**
-   * This function is called every 20 ms, no matter the mode. Use this for items like diagnostics
-   * that you want ran during disabled, autonomous, teleoperated and test.
-   *
-   * <p>This runs after the mode specific periodic functions, but before LiveWindow and
-   * SmartDashboard integrated updating.
-   */
-  @Override
-  public void robotPeriodic() {
-    // Runs the Scheduler.  This is responsible for polling buttons, adding newly-scheduled
-    // commands, running already-scheduled commands, removing finished or interrupted commands,
-    // and running subsystem periodic() methods.  This must be called from the robot's periodic
-    // block in order for anything in the Command-based framework to work.
-    CommandScheduler.getInstance().run();
-  }
-
-  /** This function is called once each time the robot enters Disabled mode. */
-  @Override
-  public void disabledInit() {}
-
-  @Override
-  public void disabledPeriodic() {}
-
-  /** This autonomous runs the autonomous command selected by your {@link RobotContainer} class. */
-  @Override
-  public void autonomousInit() {
-    m_autonomousCommand = m_robotContainer.getAutonomousCommand();
+    private Command autonomousCommand;
+    private double loopTime = 0;
+    private RobotContainer robotContainer;
+    private static Diagnostics diagnostics;
+
+
+    @Override
+    public void robotInit() {
+        if (Constants.ENABLE_LOGGING) {
+            DataLogManager.start();
+            DriverStation.startDataLog(DataLogManager.getLog(), true);
+        }
+        diagnostics = new Diagnostics();
+        robotContainer = new RobotContainer();
+        new WheelAlign(robotContainer.getDrivetrain()).schedule();
+        new ResetGyro(robotContainer.getDrivetrain(), 2).schedule();
+    }
 
-    // schedule the autonomous command (example)
-    if (m_autonomousCommand != null) {
-      m_autonomousCommand.schedule();
+    @Override
+    public void robotPeriodic() {
+        CommandScheduler.getInstance().run();
+        // Logger should stay at the end of robotPeriodic()
+        double time = (loopTime == 0) ? 0 : (Timer.getFPGATimestamp() - loopTime) * 1000;
+        Logger.logDouble("/robot/loopTime", time, Constants.ENABLE_LOGGING);
     }
-  }
 
-  /** This function is called periodically during autonomous. */
-  @Override
-  public void autonomousPeriodic() {}
+    @Override
+    public void autonomousInit() {
+        if (autonomousCommand != null) {
```

Comment:
Where is the autonomousCommand get initialized?

Reply:
should me deleted for now (not merged with autochooser yet)

Reply:
No - leave the method in place, it will not do anything if the command is null, and this way we won't forget to do that when we add the chooser.